azavea/franklin
# Move Validator typeclasses to syntax

## Overview

This PR:

- adds EO to the list of checked item extensions
- removes the `FooValidator` typeclasses in favor of syntax methods

### Checklist

- ~New tests have been added or existing tests have been modified~ it's the magic of the `stac4s` testing module :tada:

### Testing Instructions

- bring up your database: `docker-compose up -d database`
- truncate both db tables: `truncate table collections; truncate table collection_items;`
- re-import the Berlin example catalog: `bloop run application -- import --catalog-root s3://rasterfoundry-development-data-us-east-1/berlin-catalog/catalog.json --db-host=localhost`
- start the server: `./scripts/server`
- make a request for its Sentinel-2 mosaic item: `http :9090/collections/berlin/items/Berlin+Sentinel+2+Mosaic`
- check its validation results: it should have attempted to validate the `eo` extension, and there shouldn't be any errors
- more fun:
  - sync the collection to local
  - edit the `berlin-sentinel.json` item to make `eo:bands` an empty list
  - truncate your db tables again
  - import your local catalog
  - request the item again -- it should be mad that `bands` is empty

Closes #302 

Closes #326
